packages/listbox/test-suites/ListboxMixin.suite.js

```diff
@@ -127,6 +128,62 @@ export function runListboxMixinSuite(customConfig = {}) {
         expect(el.formElements[2].checked).to.be.true;
       });
 
+      it('requests update for modelValue when checkedIndex changes', async () => {
+        const el = await fixture(html`
+          <${tag} name="gender" .modelValue=${'other'}>
+            <${optionTag} .choiceValue=${'male'}></${optionTag}>
+            <${optionTag} .choiceValue=${'female'}></${optionTag}>
+            <${optionTag} .choiceValue=${'other'}></${optionTag}>
+          </${tag}>
+        `);
+        expect(el.checkedIndex).to.equal(2);
+        const spy = sinon.spy(el, 'requestUpdate');
+        el.setCheckedIndex(1);
+        expect(spy).to.have.been.calledWith('modelValue', 'other');
+      });
+
+      it('requests update for modelValue after click', async () => {
+        const el = await fixture(html`
+          <${tag} name="gender" .modelValue=${'other'}>
+            <${optionTag} .choiceValue=${'male'}></${optionTag}>
+            <${optionTag} .choiceValue=${'female'}></${optionTag}>
+            <${optionTag} .choiceValue=${'other'}></${optionTag}>
+          </${tag}>
+        `);
+        expect(el.checkedIndex).to.equal(2);
+        const spy = sinon.spy(el, 'requestUpdate');
+        el.formElements[0].click();
+        expect(spy).to.have.been.calledWith('modelValue', 'other');
+      });
+
+      it('requests update for modelValue when checkedIndex changes for multiple choice', async () => {
+        const el = await fixture(html`
+          <${tag} name="gender" multiple-choice .modelValue=${['other']}>
+            <${optionTag} .choiceValue=${'male'}></${optionTag}>
+            <${optionTag} .choiceValue=${'female'}></${optionTag}>
+            <${optionTag} .choiceValue=${'other'}></${optionTag}>
+          </${tag}>
+        `);
+        expect(el.checkedIndex).to.eql([2]);
+        const spy = sinon.spy(el, 'requestUpdate');
+        el.setCheckedIndex(1);
+        expect(spy).to.have.been.calledWith('modelValue', sinon.match.array.deepEquals(['other']));
+      });
```

Comment:
Nice. To make it really complete, I think we should also check whether it will be recognized in the `updated` callback via the `hasChanged` function. (if modelValue is an array reference, then we would not get notified if we have the default dirty check).

Reply:
Hmm so just spy the updated callback you think? I guess it's kinda needed to really test that the update is done, and not cancelled because the dirty check fails.

Reply:
Indeed, spy on updated and inspect changedProperties

Reply:
Done, was a bit annoying to assert it because my sinon matcher wouldn't match with the multiple choice updated model value because it's an array and doesn't "true equals" the expected match, but I worked around it 👍

Reply:
Turns out we don't actually need a hasChanged btw, I checked with and without and it doesn't make a difference 🤷

Reply:
That's weird. Do we change the reference to the array somewhere?

If so, we would also have a change detected on similar arrays

Reply:
I guess we do, couldn't really figure that one out..

Reply:
Maybe you can create an issue + failing test later if you find that doing setCheckedIndex = [1, 3] when 1 and 3 are already selected results in a model-value-changed :D?

Reply:
Ok, I will 👍